message-service/src/main/java/sjchat/messages/MessageService.java

```diff
@@ -140,6 +115,11 @@ public void getChat(GetChatRequest req, StreamObserver<GetChatResponse> response
 
   @Override
   public void createChat(CreateChatRequest req, StreamObserver<CreateChatResponse> responseObserver) {
+    if (req.getTitle() == null || req.getTitle().length() == 0) {
+      responseObserver.onError(new Exception("Title is empty or missing"));
+      responseObserver.onCompleted();
```

Comment:
I don't think you should have a onCompleted after onError, I've gotten errors when my code has looked like that.

Also shouldn't you return after the onError call?